src/lib/components/TextInput/TextInput.spec.tsx

```diff
@@ -15,5 +15,63 @@ describe.concurrent('Components / Text input', () => {
 
       expect(page).toBeInTheDocument();
     });
+    it('should render Icon if renderIcon prop is set', () => {
+      const page = render(<TextInput renderIcon={(style) => <HiEye className={style} />} />).getByTestId(
+        'rendered-icon',
+      );
+
+      expect(page).toBeInTheDocument();
+    });
+
+    it('should render RightIcon if renderRightIcon prop is set', () => {
+      const page = render(<TextInput renderRightIcon={(style) => <HiEye className={style} />} />).getByTestId(
+        'rendered-right-icon',
+      );
+
+      expect(page).toBeInTheDocument();
+    });
+
+    it('rendered Icon should allow for click events', () => {
+      render(
+        <TextInput
+          renderIcon={(style) => (
+            <HiEye
+              data-testid="icon"
+              className={style}
+              onClick={() => {
+                const target = document.querySelector('[data-testid="rendered-icon"]');
+                const div = document.createElement('div');
+                div.innerHTML = 'clicked';
+                target?.parentNode?.insertBefore(div, target.nextSibling);
+              }}
+            />
+          )}
+        />,
+      );
+      const icon = screen.getByTestId('icon');
+      fireEvent.click(icon);
```

Comment:
Please use the syntax seen, e.g. here https://github.com/themesberg/flowbite-react/blob/main/src/components/Accordion/Accordion.spec.tsx#L69